server/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -1605,4 +1605,71 @@ public void test_eliminate_cross_join_with_filter() throws Exception {
             "    └ Collect[doc.t2 | [y] | true]"
         );
     }
+
+    /**
+     * https://github.com/crate/crate/issues/14961
+     */
+    @Test
+    @UseRandomizedSchema(random = false)
+    @UseRandomizedOptimizerRules(0)
+    @UseHashJoins(1)
+    public void test_number_of_hash_symbol_match_lhs_rhs_with_nested_joins() throws Exception {
```

Comment:
- Do you think this is necessary, given that the next test is more complex? (Since these are integration tests and are relatively heavy)
- If you think it's important to test both variations, it's better to have more unit tests in JoinTest, since we're mainly asserting the plan, and keep only one integ test that asserts the result as well.